src-tauri/src/api/call_event.rs

```diff
@@ -72,3 +74,29 @@ pub struct Team {
     pub email: String,
     pub company_name: String,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, Error)]
+pub struct ServerApiError {
+    pub id: String,
+    pub message: String,
+    pub request_id: String,
+    pub status_code: i16,
+}
+
+impl fmt::Display for ServerApiError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+
+        write!(
+            f,
+            "{} {}message{}: {}{}{}, {}status_code{}: {}, {}request_id{}: {}{}{} {}",
+            "{",
+            r#"""#,r#"""#,
+            r#"""#, &self.message, r#"""#,
+            r#"""#,r#"""#,
+            &self.status_code,
+            r#"""#,r#"""#,
+            r#"""#,  &self.request_id,  r#"""#,
+            "}"
+        )
+    }
+}
```

Comment:
Please try use derive_more::Display

Reply:
This must be serialised to JSON, as these values are used in FE

Reply:
Display from derive_more doesn't works like Debug. It allow you to provide fmt to describe how things should be displayed.

```rust
#[derive(derive_more::Display)]
#[display(fmt = "{a} and {b}")]
struct A {
    pub a: u8,
    pub b: String,
}
```

Reply:
Also there's something called format_args! and f.write_fmt()

Reply:
That won't help because there are special characters that have to be used like `"` or `{`. FE requires JSON object to display that error, I changed it to serde json macro

Reply:
For special characters you just use `{{`, `}}` and `\"`

Reply:
Also serde_json::to_string would be much safer approach

Reply:
Using `{{}}` did not help with `{` and made the derivative display unreadable, there were too many escape characters and brackets to understand where the values were used. The `serde_json::to_string` approach is better